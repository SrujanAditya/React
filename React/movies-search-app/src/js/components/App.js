import React from 'react';
import { Component } from 'react';
import '../css/App.css';
import Header from '../ui/Header';
import Footer from '../ui/Footer';
import MoviesList from '../ui/MoviesList';
import SubHeader from '../ui/SubHeader';
import {fetchAndSetData,updateMovies,fetchAndSetMovieData} from '../actions/Actions';
import {connect} from 'react-redux';

export class App extends Component {
  constructor(props) {
    super(props)
    window.scrollTo(0,0);
  }
  render(){
    return (
      <div>
        <Header fetchData={this.props.fetchAndSetData}/> 
        {(this.props.dataLength != 0) ? 
          <SubHeader resultsFound={this.props.resultsFound} updateMovies={this.props.updateMovies}/> 
          : null 
        }
        {(this.props.dataLength != 0) ? 
          <MoviesList movies={this.props.movies} movieDes={this.props.fetchAndSetMovieData} history={this.props.history}/>
          : <div className="noFilms">No Films Found</div>
        }
        <Footer />
      </div>
    )
  }
}


export const mapStateToProps = state =>({
  dataLength : state.data.length,
  movieDataLength : state.movieData.length,
  movieData : state.movieData,
  resultsFound : state.resultsFound,
  searchText : state.searchText,
  movies : state.data
})

export const mapDispatchToProps = {
  fetchAndSetData,
  updateMovies,
  fetchAndSetMovieData
};

export default connect(mapStateToProps,mapDispatchToProps)(App);
