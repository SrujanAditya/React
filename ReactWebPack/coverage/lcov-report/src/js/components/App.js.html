<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/js/components/App.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\..\index.html">All files</a> / <a href="index.html">src/js/components</a> App.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>6/6</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>4/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>3/3</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>5/5</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';
import { Component } from 'react';
import '../css/App.css';
import Header from '../ui/Header';
import Footer from '../ui/Footer';
import MoviesList from '../ui/MoviesList';
import SubHeader from '../ui/SubHeader';
import {fetchAndSetData,updateMovies,fetchAndSetMovieData} from '../actions/Actions';
import {connect} from 'react-redux';
&nbsp;
export class App extends Component {
  constructor(props) {
    super(props)
    window.scrollTo(0,0);
  }
  render(){
    return (
      &lt;div&gt;
        &lt;Header fetchData={this.props.fetchAndSetData}/&gt; 
        {(this.props.dataLength != 0) ? 
          &lt;SubHeader resultsFound={this.props.resultsFound} updateMovies={this.props.updateMovies}/&gt; 
          : null 
        }
        {(this.props.dataLength != 0) ? 
          &lt;MoviesList movies={this.props.movies} movieDes={this.props.fetchAndSetMovieData} history={this.props.history}/&gt;
          : &lt;div className="noFilms"&gt;No Films Found&lt;/div&gt;
        }
        &lt;Footer /&gt;
      &lt;/div&gt;
    )
  }
}
&nbsp;
&nbsp;
export const mapStateToProps = state =&gt;({
  dataLength : state.data.length,
  movieDataLength : state.movieData.length,
  movieData : state.movieData,
  resultsFound : state.resultsFound,
  searchText : state.searchText,
  movies : state.data
})
&nbsp;
export const mapDispatchToProps = {
  fetchAndSetData,
  updateMovies,
  fetchAndSetMovieData
};
&nbsp;
export default connect(mapStateToProps,mapDispatchToProps)(App);
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Mon Nov 12 2018 11:30:53 GMT+0530 (India Standard Time)
</div>
</div>
<script src="..\..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\..\sorter.js"></script>
</body>
</html>
