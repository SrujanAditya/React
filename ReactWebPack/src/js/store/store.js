import appReducer from '../reducers/index.js'
import {createStore, applyMiddleware} from 'redux'; 
import thunk from 'redux-thunk';

// const initialState = (localStorage['redux-store']) ?
// 	JSON.parse(localStorage['redux-store']) :
// 	{}

// const saveState = () => {
// 	const state = JSON.stringify(store.getState())
//     localStorage['redux-store'] = state
// }

const store = createStore(appReducer,applyMiddleware(thunk));

// store.subscribe(saveState);

export default store;