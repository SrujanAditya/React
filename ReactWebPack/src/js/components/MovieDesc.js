import React from 'react';
import { Component } from 'react';
import MovieData from '../ui/MovieData';
import MoviesList from '../ui/MoviesList';
import MovieSubHeader from '../ui/MovieSubHeader';
import Footer from '../ui/Footer';
import {fetchAndSetMovieData} from '../actions/Actions';
import {connect} from 'react-redux';
import '../css/App.css'

export class MovieDesc extends Component {
    constructor(props) {
        super(props);
        this.props.fetchAndSetMovieData(props.match.params.id);
        window.scrollTo(0,0);
    }
    getSnapshotBeforeUpdate(prevProps) {
        if(prevProps.match.params.id !== this.props.match.params.id) {
            return this.props
        }
        return null;
    }
    componentDidUpdate(prevProps, prevState, snapshot) {
        if(snapshot !== null && snapshot.match.params.id) {
            this.props.fetchAndSetMovieData(snapshot.match.params.id);
            window.scrollTo(0,0);
        }
    } 
    render(){
        return (
            <div>
                <MovieData movieData={this.props.movieData} history={this.props.history}/> 
                {(this.props.movieDataLength != 0) ? 
                    <MovieSubHeader searchText={this.props.searchText}/> 
                    : null 
                }
                {(this.props.dataLength != 0) ? 
                    <MoviesList movies={this.props.movies} movieDes={this.props.fetchAndSetMovieData} history={this.props.history}/>
                    : <div className="noFilms">No Films Found</div>
                }
                <Footer /> 
            </div>
        )
    }
}

export const mapStateToProps = state =>({
    dataLength : state.data.length,
    movieDataLength : state.movieData.length,
    movieData : state.movieData,
    searchText : state.searchText,
    movies : state.data
})

export const mapDispatchToProps = {
    fetchAndSetMovieData
};
  
export default connect(mapStateToProps,mapDispatchToProps)(MovieDesc);
