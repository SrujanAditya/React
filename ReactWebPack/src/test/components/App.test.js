import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {App,mapStateToProps} from '../../js/components/App';
import {store} from '../../js/store/store';
import { Provider } from 'react-redux';

Enzyme.configure({ adapter: new Adapter() })

describe('ui', () => {
    const props = {
        dataLength : 1
    }
    it('App component renders correctly with changing conditions', () => {
        const AppComp = shallow(<App {...props}/>);
        expect(AppComp).toMatchSnapshot();
        AppComp.setProps({
            dataLength : 0
        })
        expect(AppComp).toMatchSnapshot();
    }) 
})

describe('ui', () => {
    it('App component renders correctly with store', () => {
        const AppComp = shallow( <Provider store={store}><App /></Provider> )
        expect(AppComp).toMatchSnapshot();
    }) 
})

describe('testing map state to props and map dispatch to props', () => {
    const initialState = {
        data: [],
        movieData: [],
        resultsFound: 100,
        searchText: 'kill'
    }

    it('testing map state to props : dataLength', () => {
        expect(mapStateToProps(initialState).dataLength).toEqual(0);
    });

    it('testing map state to props : movieDataLength', () => {
        expect(mapStateToProps(initialState).movieDataLength).toEqual(0);
    });

    it('testing map state to props : movieData', () => {
        expect(mapStateToProps(initialState).movieData).toEqual([]);
    });

    it('testing map state to props : resultsFound', () => {
        expect(mapStateToProps(initialState).resultsFound).toEqual(100);
    });

    it('testing map state to props : searchText', () => {
        expect(mapStateToProps(initialState).searchText).toEqual('kill');
    });

    it('testing map state to props : movies', () => {
        expect(mapStateToProps(initialState).movies).toEqual([]);
    });
})
