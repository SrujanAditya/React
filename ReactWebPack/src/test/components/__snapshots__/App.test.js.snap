// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui App component renders correctly with changing conditions 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    dataLength={1}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Header />,
        <SubHeader />,
        <MoviesList />,
        <Footer />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "fetchData": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "resultsFound": undefined,
          "updateMovies": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "history": undefined,
          "movieDes": undefined,
          "movies": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Header />,
          <SubHeader />,
          <MoviesList />,
          <Footer />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "fetchData": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "resultsFound": undefined,
            "updateMovies": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "history": undefined,
            "movieDes": undefined,
            "movies": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`ui App component renders correctly with changing conditions 2`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    dataLength={0}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Header />,
        null,
        <div
          className="noFilms"
        >
          No Films Found
        </div>,
        <Footer />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "fetchData": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      null,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "No Films Found",
          "className": "noFilms",
        },
        "ref": null,
        "rendered": "No Films Found",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Header />,
          null,
          <div
            className="noFilms"
          >
            No Films Found
          </div>,
          <Footer />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "fetchData": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        null,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "No Films Found",
            "className": "noFilms",
          },
          "ref": null,
          "rendered": "No Films Found",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`ui App component renders correctly with store 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Provider>
    <App />
  </Provider>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {},
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
